rs/boundary_node/anonymization/backend/src/lib.rs

```diff
@@ -129,8 +141,84 @@ fn timers() {
                     ps.insert(p.to_owned(), ());
                 });
             });
+
+            // Remove stale public-keys
+            PUBLIC_KEYS.with(|ks| {
+                let mut ks = ks.borrow_mut();
+
+                let stale: Vec<Principal> = ks
+                    .iter()
+                    .filter_map(|(p, _)| (!ids.contains(&p)).then_some(p))
+                    .collect();
+
+                for p in stale {
+                    ks.remove(&p);
+                }
+            });
+
+            // Remove stale encrypted values
+            ENCRYPTED_VALUES.with(|vs| {
+                let mut vs = vs.borrow_mut();
+
+                let stale: Vec<Principal> = vs
+                    .iter()
+                    .filter_map(|(p, _)| (!ids.contains(&p)).then_some(p))
+                    .collect();
+
+                for p in stale {
+                    vs.remove(&p);
+                }
+            });
+
+            // Remove stale queue entries
+            QUEUE.with(|q| {
+                let mut q = q.borrow_mut();
+
+                let stale: Vec<Principal> = q
+                    .iter()
+                    .filter_map(|(p, _)| (!ids.contains(&p)).then_some(p))
+                    .collect();
+
+                for p in stale {
+                    q.remove(&p);
+                }
+            });
         });
     });
+
+    // Leader
+    set_timer_interval(Duration::from_secs(30), || {
+        // Collect candidates
```

Comment:
Extend the comment to explain why it is not enough to just take the API BN node id list.